api/app/src/v2/Service/CourtOrderService.php

```diff
@@ -18,11 +21,121 @@ class CourtOrderService
         private readonly EntityManagerInterface $entityManager,
         private readonly LoggerInterface $logger,
     ) {
     }
 
+    private function transformDate(?string $dateValue, string $fieldName, string $format): ?string
+    {
+        if (empty($dateValue)) {
+            return null;
+        }
+
+        try {
+            $date = new \DateTimeImmutable($dateValue);
+            return $date->format($format);
+        } catch (\Exception $e) {
+            // Optional: log the error for debugging
+            error_log(sprintf(
+                "Date transform failed for %s: %s (value: %s)\n",
+                $fieldName,
+                $e->getMessage(),
+                $dateValue
+            ));
+            return null;
+        }
+    }
+
+    public function transformReportDates(array $reportArray): ?array
+    {
+        $reportArray['due_date'] = $this->transformDate($reportArray['due_date'], 'dueDate', 'Y-m-d');
+        $reportArray['submit_date'] = $this->transformDate($reportArray['submit_date'], 'submitDate', 'Y-m-d\TH:i:sP');
+        $reportArray['un_submit_date'] = $this->transformDate($reportArray['un_submit_date'], 'unSubmitDate', 'Y-m-d');
+        $reportArray['start_date'] = $this->transformDate($reportArray['start_date'], 'startDate', 'Y-m-d');
+        $reportArray['end_date'] = $this->transformDate($reportArray['end_date'], 'endDate', 'Y-m-d');
+
+        return $reportArray;
+    }
+
+    public function getCourtOrderView(string $uid, ?UserInterface $user): ?array
+    {
+        if ($user === null || !method_exists($user, 'getId')) {
+            return null;
+        }
+        $userId = (int) $user->getId();
+
+        /** @var array<int, array<string, mixed>> $courtOrderView */
+        $courtOrderView = $this->courtOrderRepository->findCourtOrderByUID($uid) ?? [];
+
+        if ($courtOrderView === []) {
+            return null;
+        }
+
+        // ===== Deputies + Authorisation Check =====
+        /** @var array<int, array<string, mixed>> $deputiesSqlResults */
+        $deputiesSqlResults = $this->deputyRepository->findDeputiesByCourtOrderUID($uid) ?? [];
+
+        $authorisedToViewCourtOrder = false;
+
+        $courtOrderView['active_deputies'] = [];
+        foreach ($deputiesSqlResults as $deputy) {
+            if (!is_array($deputy)) {
+                continue;
+            }
+            // Must have a numeric user_id to proceed
+            if (!isset($deputy['user_id']) || !is_numeric($deputy['user_id'])) {
+                continue;
+            }
+            $deputyUserId = (int) $deputy['user_id'];
+
+            // Authorisation flag
+            if ($deputyUserId === $userId) {
+                $authorisedToViewCourtOrder = true;
+            }
+
+            // Fetch deputy user details
+            /** @var array<int, array<string, mixed>> $userSqlResults */
+            $userSqlResults = $this->userRepository->findUserById($deputyUserId) ?? null;
```

Comment:
No need for a new method for this. Use `getOneOrNulLResult()` to retrieve the entity from the repository. May need to check lazy loads don't bring back lots of crud.